data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/AccumulatingBulkRequest.java

```diff
@@ -26,6 +26,10 @@
 
     int incrementAndGetRetryCount();
 
+    void setRetryCount(int count);
```

Comment:
Adding the setter will help the issue of not losing count. But, I still think that there should be a distinct class for handling the retry counts. The `AccumulatingBulkRequest` can focus on counting accumulations which is its current responsibility.

Another class like `RetryableBulkRequest` could handle the retry counting.